cmd/kube-controller-manager/app/options/persistentvolumebindercontroller.go

```diff
@@ -64,5 +72,28 @@ func (o *PersistentVolumeBinderControllerOptions) Validate() []error {
 	}
 
 	errs := []error{}
+	if _, err := ParseDialHostDenyList(o.DialHostCIDRDenylist); err != nil {
+		errs = append(errs, fmt.Errorf("Bad --dial-host-ip-denylist %w", err))
+	}
 	return errs
 }
+
+// ParseDialHostDenyList parses a comma-separated list of CIDR ranges to avoid
+// dialing from volume plugins, or `none'.  If `none', then a
+// FilteredDialOptions that allows all hosts is returned. Otherwise, a
+// FilteredDialOptions that block non-proxyable IPs along with any CIDR ranges
+// in arg is returned.
+func ParseDialHostDenyList(arg string) (*proxyutil.FilteredDialOptions, error) {
+	if arg == "none" {
+		return &proxyutil.FilteredDialOptions{DialHostCIDRDenylist: nil, DenyNonproxyableHosts: false}, nil
+	}
+	if len(arg) == 0 {
+		return &proxyutil.FilteredDialOptions{DialHostCIDRDenylist: nil, DenyNonproxyableHosts: true}, nil
+	}
```

Comment:
the distinction between these is pretty hard to follow, especially when combined with the default flag/configfile value

possible behaviors:
1. anything allowed (current behavior)
2. deny nonproxyable hosts
3. deny specific list of CIDRs
4. deny nonproxyable hosts and specific list of CIDRs

is it important to be able to be able to achieve all four of these behaviors?

possible flag-configured values:
1. no flag
2. flag with no value (`--dial-host-cidr-denylist`)
3. flag with explicit `none` value (`--dial-host-cidr-denylist=none`)
4. flag with cidr values (`--dial-host-cidr-denylist=1.2.3.4/16,...`)

flag option 2 is confusing to me

possible config file values:
1. no field
2. field with explicit null or empty list (`DialHostCIDRDenylist: null` or `DialHostCIDRDenylist: []`)
3. field with explicit `none` (`DialHostCIDRDenylist: ["none"]`)
4. field with explicit cidr values (`DialHostCIDRDenylist: ["1.2.3.4/16",...]`)

config options 1 and 2 would be treated the same by defaulting case and turned into `["none"]`... distinguishing between missing, nil, and empty lists has caused lots of confusion in the past, so we avoid it now

Reply:
I'm actually having a hard time telling what the default behavior is if someone invokes kube-controller-manager with no flag or config option provided for this... does this PR switch the default to `DenyNonproxyableHosts=true`?

Reply:
w/rt using in other components: there is no flag, but the plumbing is there, see eg pkg/controller/volume/attachdetach/attach_detach_controller.go.

w/rt flag options, it is confusing and I apologize. The behaviors we want are:

1) anything allowed (current behavior)
2) IsProxyable-like behavior onky. This can't be easily specified with a CIDR list without manually enumerating what ranges, eg, localhost map to for both IPv4 and IPv6
3) Custom CIDR ranges + IsProxyable behavior.

See ParseDialHostDenyList in cmd/kube-controller-manager/app/options/persistentvolumebindercontroller.go where things are split up.